qcengine/programs/molpro.py

```diff
@@ -68,82 +70,105 @@ def compute(self, input_data: 'ResultInput', config: 'JobConfig') -> 'Result':
-            s = "{:<4s} {:>{width}.{prec}f} {:>{width}.{prec}f} {:>{width}.{prec}f}".format(
-                sym, *geom, width=14, prec=10)
-            input_file.append(s)
-        input_file.append('}')
-
-        # Write charge and multiplicity
-        input_file.append('set,charge={}'.format(input_model.molecule.molecular_charge))
-        input_file.append('set,multiplicity={}'.format(input_model.molecule.molecular_multiplicity))
-        input_file.append('')
-
-        # Write the basis set
-        input_file.append('basis={')
-        input_file.append('default,{}'.format(input_model.model.basis))
-        input_file.append('}')
-        input_file.append('')
-
-        # Start of Molpro Commands
-        # Write energy call
-        write_hf = input_model.model.method.lower() in posthf_methods
-        if write_hf:
-            input_file.append('{HF}')
-        input_file.append('{{{:s}}}'.format(input_model.model.method))
-
-        # Write gradient call if asked for
-        if input_model.driver == 'gradient':
+        if template is None:
+            input_file = []
+            posthf_methods = {'mp2', 'ccsd', 'ccsd(t)'}
+
+            # Memory is in megawords per core for Molpro
+            memory_mw_core = int(config.memory * (1024 ** 3) / 8e6 / config.ncores)
+            input_file.append("memory,{},M".format(memory_mw_core))
+            # TODO Decide how I want this keyword to look/work
+            # if input_model.keywords.get('wfu') is not None:
+            #     input_file.append('file,2,{}'.format(input_model.keywords.get('wfu')))
             input_file.append('')
-            input_file.append('{force}')
 
-        input_file = "\n".join(input_file)
+            # Write the geom
+            xyz_block = input_model.molecule.to_string(dtype='molpro', units='Bohr')
+            input_file.append(xyz_block)
+
+            # Write the basis set
+            input_file.append('basis={')
+            input_file.append('default,{}'.format(input_model.model.basis))
+            input_file.append('}')
+            input_file.append('')
+
+            # Start of Molpro Commands
+            # Write energy call
+            energy_call = []
+            write_hf = input_model.model.method.lower() in posthf_methods
+            if write_hf:
+                energy_call.append('{HF}')
+            # TODO Support DFT calls, need to check if method is a DFT XC functional and then write {rks,XC}
+            energy_call.append('{{{:s}}}'.format(input_model.model.method))
+
+            # Write appropriate driver call
+            if input_model.driver == 'energy':
+                input_file.extend(energy_call)
+            elif input_model.driver == 'gradient':
+                input_file.extend(energy_call)
+                input_file.append('')
+                input_file.append('{force}')
+            else:
+                raise NotImplementedError('Driver {} not implemented for Molpro.'.format(input_model.driver))
```

Comment:
`NotImplementedError` not defined. should be `InputError`, I think.

Reply:
I believe `NotImplementedError` comes with the stdlib in python? Would you prefer that I used `InputError` which is defined in `qcengine/exceptions`?

Reply:
Right, we want to use all `qcengine/exceptions` where possible. This allows categorization and restart capabilities downstream along with some extra features.